packages/backend/src/core/events/EventUpdater.ts

```diff
@@ -81,73 +79,37 @@ export class EventUpdater {
     }
 
     await this.eventRepository.addMany(events)
-    this.lastProcessed = lastHour
     this.logger.info('Update completed', { timestamp: lastHour.toString() })
   }
 
-  private getFirstHour() {
-    if (this.lastProcessed) {
-      return this.lastProcessed.add(1, 'hours')
-    }
-    // it is needed for sixHourly and daily
-    // we want to avoid situation whe we ask for blockNumbers and there are no records
-    // it can happen because how logic works, it asks for multiple earlier records when needed
-    // e.g. you are at 00:00 so to calculate daily you need blockNumbers of 23 previous hourly
-    const firstHour = this.clock.getFirstHour().add(1, 'days')
-    return firstHour.isFull('day') ? firstHour : firstHour.toNext('day')
-  }
-
   async fetchRecords(
     from: UnixTime,
     to: UnixTime,
     event: EventDetails,
   ): Promise<EventRecord[]> {
-    // it is needed for sixHourly and daily
-    // generateEventRecords needs data in certain order
-    const fromAdjusted = getAdjustedFrom(from)
-
-    const eventBlockNumbers = await this.eventBlockNumbers(
-      fromAdjusted,
-      to,
-      event,
-    )
-
-    const timestamps = await this.blockNumberUpdater.getBlockRangeWhenReady(
-      fromAdjusted,
-      to,
-    )
-
-    const records = generateEventRecords(event, eventBlockNumbers, timestamps)
-
-    return records.filter((r) => r.timestamp.gte(from))
-  }
-
-  async eventBlockNumbers(
-    from: UnixTime,
-    to: UnixTime,
-    event: EventDetails,
-  ): Promise<bigint[]> {
     const fromBlock = await this.blockNumberUpdater.getBlockNumberWhenReady(
-      from.add(-1, 'hours'),
+      from,
     )
     const toBlock = await this.blockNumberUpdater.getBlockNumberWhenReady(to)
-
-    const logs = await this.ethereumClient.getLogsUsingBisection(
+    const logs = await this.ethereumClient.getAllLogs(
       event.emitter,
       event.topic,
       Number(fromBlock),
       Number(toBlock),
     )
-    return logs.map((l) => BigInt(l.blockNumber))
-  }
 
-  getLastProcessed(): UnixTime | undefined {
-    return this.lastProcessed
-  }
-}
+    const blocks = await this.blockNumberUpdater.getBlockRangeWhenReady(
+      from,
+      to,
+    )
+    const logsWithCorrespondingBlocks = findCorrespondingBlocks(blocks, logs)
+
+    const records: EventRecord[] = logsWithCorrespondingBlocks.map((l) => ({
+      name: event.name,
+      projectId: event.projectId,
+      timestamp: l.block.timestamp,
+    }))
```

Comment:
If we are only using the timestamps here then `findCorrespondingBlocks` should be renamed to `findTimestamps(blockNumbers: bigint[]): UnixTime[]`

Reply:
I can't agree. I actually spent some time refining the interface of this function.

In its current form, it's generic. It can be used not only with events (logs) but with anything really: contract deployments, or some other happening. Furthermore, it can be used to extract other things, not only timestamps from blocks. I think this way interface of the function is much clearer.